src/main/java/seedu/address/logic/parser/DeleteCommandParser.java

```diff
@@ -17,12 +18,18 @@ public class DeleteCommandParser implements Parser<DeleteCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public DeleteCommand parse(String args) throws ParseException {
-        try {
-            Index index = ParserUtil.parseIndex(args);
-            return new DeleteCommand(index);
-        } catch (ParseException pe) {
+        if (args.isEmpty()) {
             throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteCommand.MESSAGE_USAGE), pe);
+                    String.format(Messages.MESSAGE_NO_INDEX, DeleteCommand.MESSAGE_USAGE));
+        } else {
+            try {
+                Index index = parseIndex(args);
+                return new DeleteCommand(index);
+            } catch (ParseException pe) {
+                throw new ParseException(
+                        String.format(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX,
+                                DeleteCommand.MESSAGE_USAGE), pe);
+            }
```

Comment:
Maybe this section could be taken out of the `else` to reduce nesting?

If the `if` section is triggered, an exception is thrown and execution would not continue, so I don't think there is a need for an `else`